### PR TITLE
New Section on Port Selection

### PR DESCRIPTION
**Pull request by mirjak**
hopefully addresses issue #26

### CODE REVIEW

**Comment by MikeBishop** [comment]
I don't think "the same port number as would be used for the same application over TCP" is necessary general-purpose guidance.  I would argue for something more like:

> As QUIC is a general purpose transport protocol, there are no requirements that servers use a particular UDP port for all QUIC traffic.  Instead, applications which use QUIC can specify a port number to represent the combination of that application protocol and the QUIC transport, if desired.  Using the same port number that the application uses over TCP can be a useful mnemonic in some cases.

File: draft-ietf-quic-applicability.md [hunk 1]
```diff
@@ -274,6 +275,29 @@ might not be desirable. For such streams, QUIC could either provide an
 explicit interface to control prioritization, or derive the prioritization
 decision from the reliability level of the stream.
 
+# Port Selection
+
+As QUIC is a general purpose transport protocol, there are no requirements that
+servers use a particular UDP port for QUIC in general but rather the same port
```
Comment (MikeBishop) [inline draft-ietf-quic-applicability.md]: I'd put a sentence break after "...in general.  Instead, the same port number is used as...."

File: draft-ietf-quic-applicability.md [hunk 2]
```diff
@@ -274,6 +275,29 @@ might not be desirable. For such streams, QUIC could either provide an
...
+number as would be used for the same application over TCP. In case of HTTP
```
Comment (MikeBishop) [inline draft-ietf-quic-applicability.md]: "In **the** case"

File: draft-ietf-quic-applicability.md [hunk 3]
```diff
@@ -274,6 +275,29 @@ might not be desirable. For such streams, QUIC could either provide an
...
+the expectation is that port 443 is used which has already been register for
```
Comment (MikeBishop) [inline draft-ietf-quic-applicability.md]: "is used**,** which has already been register**ed**...."

File: draft-ietf-quic-applicability.md [hunk 4]
```diff
@@ -274,6 +275,29 @@ might not be desirable. For such streams, QUIC could either provide an
...
+"http protocol over TLS/SSL". However, {{QUIC-HTTP}} also specifies the use
+of Alt-Svc for HTTP/QUIC discovery which allows the server to use and
+announce of a different port number.
```
Comment (MikeBishop) [inline draft-ietf-quic-applicability.md]: "announce ~~of~~ a"

File: draft-ietf-quic-applicability.md [hunk 5]
```diff
@@ -274,6 +275,29 @@ might not be desirable. For such streams, QUIC could either provide an
...
+
+In general port numbers serves two purposes: "first, they provide a
```
Comment (MikeBishop) [inline draft-ietf-quic-applicability.md]: "In general**,**"

File: draft-ietf-quic-applicability.md [hunk 6]
```diff
@@ -274,6 +275,29 @@ might not be desirable. For such streams, QUIC could either provide an
...
+demultiplexing identifier to differentiate transport sessions between the
+same pair of endpoints, and second, they may also identify the application
+protocol and associated service to which processes connect" {{RFC6335}}.
+However, {{RFC6335}} also acknowledges that the assumption that an application
+can be identified in the network based on the port number is less true today,
+due to encapsulation, mechanism for dynmaic port assignments as well as NATs.
```
Comment (MikeBishop) [inline draft-ietf-quic-applicability.md]: "...less true today due to encapsulation, mechanism**s** for **dynamic** port assignments, and NATs."

File: draft-ietf-quic-applicability.md [hunk 7]
```diff
@@ -274,6 +275,29 @@ might not be desirable. For such streams, QUIC could either provide an
...
+
+However, whenever a none "standard" port is used ,that does not enable easy
```
Comment (MikeBishop) [inline draft-ietf-quic-applicability.md]: non~~e~~**-**standard
"used which"

File: draft-ietf-quic-applicability.md [hunk 8]
```diff
@@ -274,6 +275,29 @@ might not be desirable. For such streams, QUIC could either provide an
...
+
+In general, port numbers serves two purposes: "first, they provide a
+demultiplexing identifier to differentiate transport sessions between the
+same pair of endpoints, and second, they may also identify the application
+protocol and associated service to which processes connect" {{RFC6335}}.
+However, {{RFC6335}} also acknowledges that the assumption that an application
+can be identified in the network based on the port number is less true today,
+due to encapsulation, mechanisms for dynamic port assignments as well as NATs.
+
+However, whenever a none-standard port is used which does not enable easy
```
Comment (britram) [inline draft-ietf-quic-applicability.md]: typofix:

```suggestion
However, whenever a non-standard port is used which does not enable simple
```

File: draft-ietf-quic-applicability.md [hunk 9]
```diff
@@ -31,6 +31,7 @@ author:
 
 normative:
   RFC2119:
+  RFC6335:
```
Comment (britram) [inline draft-ietf-quic-applicability.md]: no longer necessary, RFC references can happen inline:
```suggestion
```

File: draft-ietf-quic-applicability.md [hunk 10]
```diff
@@ -274,6 +275,29 @@ might not be desirable. For such streams, QUIC could either provide an
...
+number is used as would be used for the same application over TCP. In the case
+of HTTP the expectation is that port 443 is used, which has already been
+registered for "http protocol over TLS/SSL". However, {{QUIC-HTTP}} also
+specifies the use of Alt-Svc for HTTP/QUIC discovery which allows the server to
+use and announce a different port number.
+
+In general, port numbers serves two purposes: "first, they provide a
+demultiplexing identifier to differentiate transport sessions between the
+same pair of endpoints, and second, they may also identify the application
+protocol and associated service to which processes connect" {{RFC6335}}.
```
Comment (britram) [inline draft-ietf-quic-applicability.md]: make this explicitly informative:
```suggestion
protocol and associated service to which processes connect" {{?RFC6335}}.
```

File: draft-ietf-quic-applicability.md [hunk 11]
```diff
@@ -274,6 +274,29 @@ might not be desirable. For such streams, QUIC could either provide an
...
+
+In general, port numbers serves two purposes: "first, they provide a
+demultiplexing identifier to differentiate transport sessions between the
+same pair of endpoints, and second, they may also identify the application
+protocol and associated service to which processes connect" {{!RFC6335}}.
+However, {{!RFC6335}} also acknowledges that the assumption that an application
+can be identified in the network based on the port number is less true today,
+due to encapsulation, mechanisms for dynamic port assignments as well as NATs.
+
+However, whenever a non-standard port is used which does not enable easy
```
Comment (dtikhonov) [inline draft-ietf-quic-applicability.md]: Two sentences in a row begin with "however."

File: draft-ietf-quic-applicability.md [hunk 12]
```diff
@@ -274,6 +274,29 @@ might not be desirable. For such streams, QUIC could either provide an
...
+mapping to a registered service name, this can lead to blocking by
+network elements such as firewalls that rely on the port number as a first order
+of filtering.
```
Comment (nibanks) [inline draft-ietf-quic-applicability.md]: With QUIC, multiple different application layers can easily be multiplexed on the same port, by ALPN. For instance, all the different services on a single server could be run over UDP 443. Should this be called out in this section?

Comment (britram) [reply draft-ietf-quic-applicability.md]: I think this is a related point -- is there anything we should say in general about ALPN in this document? Probably. Prefer to file a new issue and land this one now, though...

**Pull request merged**